### PR TITLE
Started complex plugin integration docs

### PR DESCRIPTION
**Pull request by evildmp**
[ci skip]

### Summary

Improves documentation for complex plugin integration.

### CODE REVIEW

File: docs/topics/complex_plugins.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+The textarea is then saved in the normal way; if there are any plugins contained in it they will be saved to the
+database before the operation can finally be considered to have concluded successfully.
+
+.. todo: the saving of the plugins in the text editor is done in Python, not JS, correct?
+
+.. todo:
+
+    If the modal's Save invokes form's submit in such a way that it can successfully run the JS validation on form
+    elements, why does it not also invoke the JS that an editor plugin uses to transfer the editor content back to the
+    textarea? It's now not clear to me why it's necessary for the CKEditor to do this separately in
```
Comment (vxsx) [inline docs/topics/complex_plugins.rst]: the code you reference is not related to saving the TextPlugin in any way
this is for _children_ of the TextPlugin, which we display as ckeditor widgets versus doing what CMS normally does and reloading the page

File: docs/topics/complex_plugins.rst [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+This plugin uses a Django form :class:`Textarea widget <django:django.forms.Textarea>`, which creates an HTML
+``<textarea>`` - *which it hides*. It then creates its own text-editing interface, the familiar CKEditor, and on
+**Save** parses the content in that (some of which may not be text content, such as plugins) and places it in the
+textarea.
+
+The textarea is then saved in the normal way; if there are any plugins contained in it they will be saved to the
+database before the operation can finally be considered to have concluded successfully.
+
+.. todo: the saving of the plugins in the text editor is done in Python, not JS, correct?
```
Comment (vxsx) [inline docs/topics/complex_plugins.rst]: yes it is.

the textarea value is cleaned up by ckeditor immediately before submission and then is parsed on the backend side. everything in between `<cms-plugin>` tags is removed and then in the live mode is replaced by appropriate plugin contents.

however the fun part starts when you create a child plugin in ckeditor but delete it before you save the TextPlugin. in this case the plugin was created but couldn't be removed - in that case the JS makes a separate request to delete that orphaned plugin

File: docs/topics/complex_plugins.rst [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+How wrapping is implemented
+***************************
+
+In order to achieve this, django CMS refers the **Save** and **Delete** actions to the admin form.
+
+Hitting django CMS's **Save** button invokes the ``<input>`` on the admin form, as though it had been invoked in the
+browser by the user, by assigning its own **Save** button to the form's ``submit`` event.
+
+.. todo: Anything else we should note?
```
Comment (vxsx) [inline docs/topics/complex_plugins.rst]: probably shouldn't, but here are few more things we do:

the cms does some hidden things like this for example https://github.com/divio/django-cms/blob/release/3.4.x/cms/static/cms/js/modules/cms.modal.js#L820-L849

or for example there are certain conditions for the modal to be closed after "saving" that may seem non-obvious https://github.com/divio/django-cms/blob/release/3.4.x/cms/static/cms/js/modules/cms.modal.js#L972-L979

or that we inject ctrl+enter shortcut to submit the modal from the keyboard even if you're not inside the form field (cmd+enter for mac)

**Comment by FinalAngel** [comment]
@evildmp I'm closing this for now. If it's still relevant please re-open with the necessary adaptions.

**Pull request closed without merging**